src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -50,7 +80,17 @@ public override async Task OnConnectedAsync(HubConnectionContext connection)
                 try
                 {
                     InitializeHub(hub, connection);
-                    await hub.OnConnectedAsync();
+
+                    if (_onConnectedMiddleware != null)
+                    {
+                        var methodInfo = typeof(THub).GetMethod("OnConnectedAsync");
```

Comment:
This is pretty weird. We should make a new context object for OnConnected and OnDisconnected

Reply:
Just need to fix these calls.